scripts/dts/edtlib.py

```diff
@@ -179,22 +185,33 @@ def get_node(self, path):
         except DTError as e:
             _err(e)
 
-    def chosen_node(self, name):
-        """
-        Returns the Node pointed at by the property named 'name' in /chosen, or
-        None if the property is missing
-        """
+    @property
+    def chosen_nodes(self):
+        ret = OrderedDict()
+
         try:
             chosen = self._dt.get_node("/chosen")
         except DTError:
-            # No /chosen node
-            return None
+            return ret
 
-        if name not in chosen.props:
-            return None
+        for name, prop in chosen.props.items():
+            try:
+                node = prop.to_path()
+            except DTError:
+                # DTS value is not phandle or string, or path doesn't exist
+                continue
+
+            if node in self._node2enode:
+                ret[name] = self._node2enode[node]
+
+        return ret
 
-        # to_path() checks that the node exists
-        return self._node2enode[chosen.props[name].to_path()]
+    def chosen_node(self, name):
```

Comment:
I'd skip this function, since it's trivial to implement.

Reply:
What do you mean? This function was here before and it's still needed by the legacy script.

Reply:
Could patch the callers to do

```python
edt.chosen_nodes.get("foo")
```

instead of

```python
edt.chosen_node("foo")
```

Nice to not have functions that are overly trivial I think. Makes it easier to find all the spots where `chosen_nodes` is used too.

Reply:
Yes, we could, but I am trying to limit changes to gen_legacy_defines.py to avoid regressions. Basically trying to freeze that file as mentioned in the top level comment:

```
# This script is similar to gen_defines.py, but is for the legacy
# macro syntax used in e.g. Zephyr 2.2.
#
# It should be considered frozen code.
```

Making this change will in turn cause other "cleanups" to this file to be the next logical course of action (https://gist.github.com/mbolivar-nordic/6dc036905ba9ebb0cb956b5861b00725), because once you have a way to get all the chosen nodes, it's wasted cycles to recompute them repeatedly.

So I'm wondering why we can't just remove this method once we remove gen_legacy_defines.py?